test/BCModel_TEST.cxx

```diff
@@ -121,11 +121,47 @@ class BCModelTest :
 
         m.MarginalizeAll();
 
-        // basic test: no segfault
-        GaussModel m2 = m;
+        // histograms are dynamically created, they are the criticial part
+        // during copying.
+        TH1* const h = m.GetMarginalizedHistogram(0);
+
+        // test copy ctor
+        {
+            GaussModel m2(m);
+
+            // non-default values should be taken over
+            TEST_CHECK_EQUAL(m2.GetNChains(), m.GetNChains());
+
+            // Original values should be untouched
+            TEST_CHECK_EQUAL(h, m.GetMarginalizedHistogram(0));
```

Comment:
A test should fully test---not just the 0'th member, but all. (Add a loop.)

Reply:
Actually this model had just one parameter. I extended to 2 dimensions and check all marginals